app/script/entity/message/File.js

```diff
@@ -0,0 +1,176 @@
+  }
+
+  /**
+   * Loads and decrypts otr asset preview
+   *
+   * @returns {Promise} Returns a promise that resolves with the asset as blob
+   */
+  load_preview() {
+    if (this.preview_resource()) {
+      return this.preview_resource().load();
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Loads and decrypts otr asset
+   *
+   * @returns {Promise} Returns a promise that resolves with the asset as blob
+   */
+  load() {
+    this.status(z.assets.AssetTransferState.DOWNLOADING);
+
+    if (this.original_resource()) {
+      this.original_resource().load()
+        .then((blob) => {
+          this.status(z.assets.AssetTransferState.UPLOADED);
+          return blob;
+        })
+        .catch((error) => {
+          this.status(z.assets.AssetTransferState.UPLOADED);
+          throw error;
+        });
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Loads and decrypts otr asset as initiates download
+   *
+   * @returns {Promise} Returns a promise that resolves with the asset as blob
+   */
+  download() {
+    if (this.status() !== z.assets.AssetTransferState.UPLOADED) {
+      return;
+    }
+
+    const download_started = Date.now();
+    const tracking_data = {
+      size_bytes: this.file_size,
+      size_mb: z.util.bucket_values((this.file_size / 1024 / 1024), [0, 5, 10, 15, 20, 25]),
+      type: z.util.get_file_extension(this.file_name),
+    };
+    amplify.publish(z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.FILE.DOWNLOAD_INITIATED, tracking_data);
+
+    return this.load()
+      .then((blob) => {
+        return z.util.download_blob(blob, this.file_name);
+      })
+      .then(() => {
+        const download_duration = (Date.now() - download_started) / 1000;
+        this.logger.info(`Downloaded asset in ${download_duration} seconds`);
+        return amplify.publish(z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.FILE.DOWNLOAD_SUCCESSFUL,
+          $.extend(tracking_data, {time: download_duration}));
+      })
+      .catch((error) => {
+        this.logger.error('Failed to download asset', error);
+        return amplify.publish(z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.FILE.DOWNLOAD_FAILED, tracking_data);
+      });
+  }
+
+  cancel_download() {
+    this.status(z.assets.AssetTransferState.UPLOADED);
+
+    if (this.original_resource()) {
+      return this.original_resource().cancel_download();
+    }
+
+    return undefined;
+  }
+
+  cancel(message_et) {
+    if (typeof this.upload_cancel === 'function') {
+      this.upload_cancel();
+    }
+    amplify.publish(z.event.WebApp.CONVERSATION.ASSET.CANCEL, message_et);
+    return amplify.publish(z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.FILE.UPLOAD_CANCELLED, {
+      size_bytes: this.file_size,
+      size_mb: z.util.bucket_values((this.file_size / 1024 / 1024), [0, 5, 10, 15, 20, 25]),
+      type: z.util.get_file_extension(this.file_name),
+    }
+    );
+  }
+
+  reload() {
+    return this.logger.info('Restart upload');
```

Comment:
What's this? Do not return but not implemented or not needed for this type of asset.